packages/core/src/api/guild.ts

```diff
@@ -0,0 +1,869 @@
+		options: RESTPutAPIGuildBanJSONBody = {},
+		reason?: string,
+	) {
+		await this.rest.put(Routes.guildBan(guildId, userId), { reason, body: options });
+	}
+
+	/**
+	 * Unbans a user from a guild
+	 *
+	 * @param guildId - The id of the guild to unban the member in
+	 * @param userId - The id of the user to unban
+	 * @param reason - The reason for unbanning the user
+	 */
+	public async unbanUser(guildId: Snowflake, userId: Snowflake, reason?: string) {
+		await this.rest.delete(Routes.guildBan(guildId, userId), { reason });
+	}
+
+	/**
+	 * Gets all the roles in a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the roles fo
+	 */
+	public async getRoles(guildId: Snowflake) {
+		return this.rest.get(Routes.guildRoles(guildId)) as Promise<RESTGetAPIGuildRolesResult>;
+	}
+
+	/**
+	 * Creates a guild role
+	 *
+	 * @param guildId - The id of the guild to create the role in
+	 * @param data - The data to create the role with
+	 * @param reason - The reason for creating the role
+	 */
+	public async createRole(guildId: Snowflake, data: RESTPostAPIGuildRoleJSONBody, reason?: string) {
+		return this.rest.post(Routes.guildRoles(guildId), { reason, body: data }) as Promise<RESTPostAPIGuildRoleResult>;
+	}
+
+	/**
+	 * Sets a role position in a guild
+	 *
+	 * @param guildId - The id of the guild to set role positions for
+	 * @param data - The data for setting a role position
+	 * @param reason - The reason for setting the role position
+	 */
+	public async setRolePosition(guildId: Snowflake, data: RESTPatchAPIGuildRolePositionsJSONBody, reason?: string) {
+		return this.rest.patch(Routes.guildRoles(guildId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildRolePositionsResult>;
+	}
+
+	/**
+	 * Edits a guild role
+	 *
+	 * @param guildId - The id of the guild to edit the role in
+	 * @param roleId - The id of the role to edit
+	 * @param data - data for editing the role
+	 * @param reason - The reason for editing the role
+	 */
+	public async editRole(guildId: Snowflake, roleId: string, data: RESTPatchAPIGuildRoleJSONBody, reason?: string) {
+		return this.rest.patch(Routes.guildRole(guildId, roleId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildRoleResult>;
+	}
+
+	/**
+	 * Deletes a guild role
+	 *
+	 * @param guildId - The id of the guild to delete the role in
+	 * @param roleId - The id of the role to delete
+	 * @param reason - The reason for deleting the role
+	 */
+	public async deleteRole(guildId: Snowflake, roleId: string, reason?: string) {
+		await this.rest.delete(Routes.guildRole(guildId, roleId), { reason });
+	}
+
+	/**
+	 * Edits the multi-factor-authentication (MFA) level of a guild
+	 *
+	 * @param guildId - The id of the guild to edit the MFA level for
+	 * @param level - The new MFA level
+	 * @param reason - The reason for editing the MFA level
+	 */
+	public async editMFALevel(guildId: Snowflake, level: number, reason?: string) {
+		return this.rest.post(Routes.guild(guildId), {
```

Comment:
Post? Did you mean patch?

Reply:
No, it is post. But the route is not correct
https://discord.com/developers/docs/resources/guild#modify-guild-mfa-level